lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
Sounds good to me then! Another thing we could consider would be some way of adding those flags to the default `compilers.yaml` file, but that would probably require more work to get working.

Once the new rebased PR is merged, I might do something similar for NAG. The NAG compilers are very picky about good Fortran code, so most Spack packages won't compile unless you use `-mismatch -dusty` to turn off some of those warnings.